master/buildbot/config.py

```diff
@@ -827,6 +828,11 @@ def __init__(self, name=None, slavename=None, slavenames=None,
         self.nextSlave = nextSlave
         if nextSlave and not callable(nextSlave):
             error('nextSlave must be a callable')
+            # Keeping support of the previous nextSlave API
+        if nextSlave and (nextSlave.func_code.co_argcount == 2 or
+                          (isinstance(nextSlave, MethodType) and
+                           nextSlave.func_code.co_argcount == 3)):
```

Comment:
Actually, when I gave a look at the coverage report before adding the `# pragma: no cover`, the `if` statement was covered but not was the `lambda` statement.

However, this use case is covered by unit tests (test_process_buildrequestdistributor), where **nextSlave** functions taking only 2 parameters pass.
 
I even added a more explicit one in this patch :)

That's why, I put it on the line not covered, as if I had set it on the top block it would have deactivated the covered part...